display-guild.lic

```diff
@@ -0,0 +1,91 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#display-guild
+=end
+
+no_pause_all
+no_kill_all
+
+arg_definitions = [
+    [
+	{ name: 'debug', regex: /debug/i, optional: true, description: 'Very verbose, may cause lich to hang for several seconds if using large contact files' }
+	]
+]
+args = parse_args(arg_definitions)
+$debug_mode_cc = args.debug
+
+Contacts = YAML.load_file("./scripts/data/base-contacts.yaml")
+Contacts = Contacts['contacts']
+
+#DEBUG INFO
+echo "---Base Contacts---" if $debug_mode_cc
+echo Contacts.inspect if $debug_mode_cc
+echo Contacts.class if $debug_mode_cc
+
+#Check for <CharName>-contacts.yaml file and merges it if exists
+if File.exist?("./scripts/profiles/#{checkname}-contacts.yaml")
+	echo "Personal contact file found - now importing." if $debug_mode_cc
+	
+	Personal_Contacts = YAML.load_file("./scripts/profiles/#{checkname}-contacts.yaml")
+	Personal_Contacts = Personal_Contacts['contacts']
+
+	#DEBUG INFO
+	echo "---Personal Contacts---" if $debug_mode_cc
+	echo Personal_Contacts.inspect if $debug_mode_cc
+	echo Personal_Contacts.class if $debug_mode_cc
+
+	Contacts.merge!(Personal_Contacts)
+	Names = Contacts.map {|row| row[0]}
+	echo "***Successfully loaded " + Names.length.to_s + " contacts."
+	
+	#DEBUG INFO
+	echo "---Names Array---" if $debug_mode_cc
+	echo Names.inspect if $debug_mode_cc
+	echo Names.class if $debug_mode_cc
+end
+
+#DEBUG INFO
+echo "---Merged Contacts---" if $debug_mode_cc
+echo Contacts.inspect if $debug_mode_cc
+echo Contacts.class if $debug_mode_cc
+echo "-----" if $debug_mode_cc
+
+class Stalker
+  @@subs = {}
+
+  def self.subs
+    @@subs
+  end
+
+  def self.clear
+    @@subs = {}
+  end
+
+  def self.add(target, result)
+    @@subs[/#{target}/] = result
+  end
+end
+
+action = proc do |client_string|
+  matches = Stalker.subs.select { |key, _| key =~ client_string }
+  matches.each { |target, result| client_string.gsub!(target, result) }
+  client_string
+end
+
+DownstreamHook.remove('stalker')
+DownstreamHook.add('stalker', action)
+
+before_dying do
+  DownstreamHook.remove('stalker')
+end
+
+Names.each do |name|
+	if (Contacts[name]['guild'] == 'Thief')
+		Contacts[name]['guild'] = 'X'
+	end
+	if (Contacts[name]['guild'] == 'Bard')
+		Contacts[name]['guild'] = 'D'
+	end
+	Stalker.add((name.capitalize),Contacts[name]['guild'].chr+':'+name.capitalize)

```

Comment:
This is another place string interpolation could be used (`"#{Contacts[name]['guild'].chr}:#{name.capitalize}"`)